bug: Freshly installed @rehearsal/cli cannot find module 'typescript'

Error appears when running globally installed `@rehearsal/cli` without typescript:
 
```
everbits@everbits-mn2 ~ % rehearsal -v
node:internal/modules/cjs/loader:998
  throw err;
  ^

Error: Cannot find module 'typescript'
Require stack:
- .../tools/image/packages/@rehearsal/cli/lib/node_modules/@rehearsal/reporter/dist/src/reporter.js
- .../tools/image/packages/@rehearsal/cli/lib/node_modules/@rehearsal/reporter/dist/src/index.js
- .../tools/image/packages/@rehearsal/cli/lib/node_modules/@rehearsal/cli/dist/src/commands/migrate/tasks/convert.js
- .../tools/image/packages/@rehearsal/cli/lib/node_modules/@rehearsal/cli/dist/src/commands/migrate/tasks/index.js
- .../tools/image/packages/@rehearsal/cli/lib/node_modules/@rehearsal/cli/dist/src/commands/migrate/index.js
- .../tools/image/packages/@rehearsal/cli/lib/node_modules/@rehearsal/cli/dist/src/index.js
- .../tools/image/packages/@rehearsal/cli/lib/node_modules/@rehearsal/cli/bin/rehearsal.js
- 
```

Seems we have to move `typescript`dependency from `devDependencies` and `peerDependencies` to `dependencies` in [@rehearsal/cli](https://github.com/rehearsal-js/rehearsal-js/blob/master/packages/cli/package.json#L53-L60) because it's required for a normal workflow.

--- a/packages/cli/src/commands/migrate/tasks/validate.ts
+++ b/packages/cli/src/commands/migrate/tasks/validate.ts
@@ -1,10 +1,58 @@
 import { resolve } from 'path';
 import { ListrTask } from 'listr2';
 import { Logger } from 'winston';
-import { existsSync } from 'fs-extra';
+import { existsSync, readFileSync } from 'fs-extra';
 
 import { getEsLintConfigPath } from '../../../utils';
 import type { MigrateCommandContext, MigrateCommandOptions } from '../../../types';
+
+function checkLintConfig(basePath: string, logger: Logger): boolean {
+  const lintConfigPath = getEsLintConfigPath(basePath);
+  if (!lintConfigPath) {
+    logger.warn(
+      `Eslint config (.eslintrc.{js,yml,json,yaml}) does not exist. You need to run rehearsal migrate first before you can run rehearsal migrate --regen`
+    );
+    return false;
+  }
+  return true;
+}
+
+function checkTsConfig(basePath: string, logger: Logger): boolean {
+  const tsConfigPath = resolve(basePath, 'tsconfig.json');
+  if (!existsSync(tsConfigPath)) {
+    logger.warn(
+      `${tsConfigPath} does not exist. You need to run rehearsal migrate first before you can run rehearsal migrate --regen`
+    );
+    return false;
+  }
+  return true;
+}
+
+function checkPackageJson(basePath: string): boolean {
+  const packageJsonPath = resolve(basePath, 'package.json');
+  if (!existsSync(packageJsonPath)) {
+    throw new Error(
+      `${packageJsonPath} does not exists. Please run rehearsal migrate inside a project with a valid package.json.`
+    );
+  }
+  return true;
+}
+
+function checkGitIgnore(basePath: string): boolean {
+  const gitignorePath = resolve(basePath, '.gitignore');
+  if (!existsSync(gitignorePath)) {
+    return true;
+  }
+
+  const gitignore = readFileSync(gitignorePath, 'utf-8');
+  const rehearsalRegex = /\.rehearsal.*/g;
+  if (rehearsalRegex.test(gitignore)) {
+    throw new Error(
+      `.rehearsal directory is ignored by .gitignore file. Please remove it from .gitignore file and try again.`
+    );
+  }
+  return true;
+}
 
 export async function validateTask(
   options: MigrateCommandOptions,
@@ -14,25 +62,11 @@ export async function validateTask(
     title: 'Validate project',
     enabled: (ctx: MigrateCommandContext): boolean => !ctx.skip,
     task: async (): Promise<void> => {
-      const packageJsonPath = resolve(options.basePath, 'package.json');
-      if (!existsSync(packageJsonPath)) {
-        throw new Error(
-          `${packageJsonPath} does not exists. Please run rehearsal migrate inside a project with a valid package.json.`
-        );
-      }
+      checkPackageJson(options.basePath);
+      checkGitIgnore(options.basePath);
       if (options.regen) {
-        const lintConfigPath = getEsLintConfigPath(options.basePath);
-        if (!lintConfigPath) {
-          logger.warn(
-            `Eslint config (.eslintrc.{js,yml,json,yaml}) does not exist. You need to run rehearsal migrate first before you can run rehearsal migrate --regen`
-          );
-        }
-        const tsConfigPath = resolve(options.basePath, 'tsconfig.json');
-        if (!existsSync(tsConfigPath)) {
-          logger.warn(
-            `${tsConfigPath} does not exist. You need to run rehearsal migrate first before you can run rehearsal migrate --regen`
-          );
-        }
+        checkLintConfig(options.basePath, logger);
+        checkTsConfig(options.basePath, logger);
       }
     },
   };

--- a/packages/cli/test/commands/migrate/validate.test.ts
+++ b/packages/cli/test/commands/migrate/validate.test.ts
@@ -1,5 +1,5 @@
 import { resolve } from 'path';
-import { createFileSync, rmSync } from 'fs-extra';
+import { createFileSync, rmSync, writeFileSync } from 'fs-extra';
 import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
 import { createLogger, format, transports } from 'winston';
 
@@ -46,7 +46,7 @@ describe('Task: validate', async () => {
     vi.clearAllMocks();
   });
 
-  test('pass with paackage.json', async () => {
+  test('pass with package.json', async () => {
     const options = createMigrateOptions(basePath);
     const tasks = [await validateTask(options, logger)];
 
@@ -61,6 +61,19 @@ describe('Task: validate', async () => {
     rmSync(resolve(basePath, 'package.json'));
 
     await expect(() => listrTaskRunner(tasks)).rejects.toThrowError(`package.json does not exists`);
+  });
+
+  test('error if .gitignore has .rehearsal', async () => {
+    const options = createMigrateOptions(basePath);
+    const tasks = [await validateTask(options, logger)];
+
+    const gitignore = `.rehearsal\nfoo\nbar`;
+    const gitignorePath = resolve(basePath, '.gitignore');
+    writeFileSync(gitignorePath, gitignore, 'utf-8');
+
+    await expect(() => listrTaskRunner(tasks)).rejects.toThrowError(
+      `.rehearsal directory is ignored by .gitignore file. Please remove it from .gitignore file and try again.`
+    );
   });
 
   test('show warning message for missing files in --regen', async () => {

